config/config-api/src/main/java/com/thoughtworks/go/config/EnvironmentVariableConfig.java

```diff
@@ -313,16 +313,16 @@ public void deserialize(String name, String value, boolean isSecure, String encr
         setName(name);
         setIsSecure(isSecure);
 
-        if (!isSecure && encryptedValue != null) {
+        if (!isSecure && StringUtils.isNotEmpty(encryptedValue)) {
             errors().add(ENCRYPTEDVALUE, "You may specify encrypted value only when option 'secure' is true.");
         }
 
-        if (value != null && encryptedValue != null) {
+        if (StringUtils.isNotEmpty(value) && StringUtils.isNotEmpty(encryptedValue)) {
             addError("value", "You may only specify `value` or `encrypted_value`, not both!");
             addError(ENCRYPTEDVALUE, "You may only specify `value` or `encrypted_value`, not both!");
         }
 
-        if (encryptedValue != null) {
+        if (StringUtils.isNotEmpty(encryptedValue)) {
```

Comment:
[`StringUtils.isNotEmpty`](http://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isNotEmpty%28java.lang.String%29) will return `true` for " " strings as well. [`StringUtils.isNotBlank`](http://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isNotBlank%28java.lang.String%29) might be a better choice. WDYT?